### PR TITLE
Add options for all codegen plugins

### PR DESCRIPTION
**Pull request by KyleFromKitware**
Codegen plugins can be enabled or disabled individually by setting
gRPC_BUILD_\<PLUGIN\> to ON or OFF.

Fixes: #15949

@yang-g
@zackgalbreath

### CODE REVIEW

File: templates/CMakeLists.txt.template [hunk 1]
```diff
@@ -264,13 +264,18 @@
     endforeach()
   endfunction()
 
-  add_custom_target(plugins
-    DEPENDS
+  set(_gRPC_PLUGIN_LIST)
   % for tgt in targets:
   % if tgt.build == 'protoc':
-    ${tgt.name}
+  option(gRPC_BUILD_${tgt.name.upper()} "Build ${tgt.name}" ON)
```
Comment (jtattermusch) [inline templates/CMakeLists.txt.template]: nit: worth adding a comment? It's hard to guess for someone who doesn't already know.

File: templates/CMakeLists.txt.template [hunk 2]
```diff
@@ -356,18 +361,18 @@
   % for tgt in targets:
   % if tgt.build in ["all", "protoc", "tool", "test", "private"] and not tgt.boringssl:
   % if tgt.build in ["test", "private"]:
-  if (gRPC_BUILD_TESTS)
+  if(gRPC_BUILD_TESTS)
```
Comment (jtattermusch) [inline templates/CMakeLists.txt.template]: was removing the space intended? If not, let's stay consistent with the rest of the file.

Comment (KyleFromKitware) [reply templates/CMakeLists.txt.template]: Yes, that was intentional.

File: templates/CMakeLists.txt.template (hunk 1 above)
Comment (KyleFromKitware) [reply templates/CMakeLists.txt.template]: Add a comment in the code? Or to the documentation? Where?

File: templates/CMakeLists.txt.template (hunk 2 above)
Comment (jtattermusch) [reply templates/CMakeLists.txt.template]: It the original formatting against cmake code style? If so, can we do things like updating formatting in a separate PR (and do that consistently everywhere, if it's important).

Comment (KyleFromKitware) [reply templates/CMakeLists.txt.template]: It is indeed not the recommended style, though obviously there are many projects that use it. I thought it was a mistake and did not realize it was intentional. However, duplicating the condition in the `endif()` block is even *more* not-recommended... is it OK if I at least get rid of that for this particular `if()` block?

File: templates/CMakeLists.txt.template (hunk 1 above)
Comment (KyleFromKitware) [reply templates/CMakeLists.txt.template]: Yes, I agree, I will add a comment explaining it.

File: templates/CMakeLists.txt.template (hunk 2 above)
Comment (jtattermusch) [reply templates/CMakeLists.txt.template]: sg, but please file an issue with the list of cmake style problems that we currently have in our CMakeLists.txt?
also, is there some good linter for cmake files?

Comment (KyleFromKitware) [reply templates/CMakeLists.txt.template]: > also, is there some good linter for cmake files?

Not that I know of :( we've talked about writing one, but I don't know if any real progress has been made on it.

**Comment by KyleFromKitware** [comment]
This is now blocked by #20765.

**Comment by jtattermusch** [comment]
@KyleFromKitware  https://github.com/grpc/grpc/pull/20765 has been merged.

**Comment by KyleFromKitware** [comment]
OK, I've fixed the merge conflicts.

**Comment by jtattermusch** [comment]
Known failures: https://github.com/grpc/grpc/issues/20694

**Pull request merged**